paper-api/src/main/java/org/bukkit/inventory/RecipeChoice.java

```diff
@@ -35,6 +37,46 @@ public interface RecipeChoice extends Predicate<ItemStack>, Cloneable {
     }
     // Paper end
 
+    /**
+     * Creates a choice that will be valid only if one of the stacks is
+     * exactly matched (aside from stack size).
+     *
+     * @param stacks the ItemStacks to match against.
+     *               Cannot be empty or contain empty/air stacks.
+     * @return a new ExactChoice.
+     */
+    static @NotNull ExactChoice exactChoice(@NotNull ItemStack... stacks) {
```

Comment:
Lets avoid upstream's mistake. For situations where it's not valid to pass 0 values, and you want a varargs parameter, you can do smth like.
```suggestion
    static @NotNull ExactChoice exactChoice(@NotNull ItemStack first, ItemStack... others) {
```
That forces them to supply at least one value, and optionally anymore